README.md

```diff
@@ -26,6 +29,182 @@ to run `zosma`.
+
+```java
+Flux.from(chessGamePublisher)
+    .transform(new ChessFormationDeducer())
+```
+
+#### Parallelizing deducers
+
+At any point during data processing it might be desirable to parallelize certain computations and
+combine their results. `zosma` contains a `ParallelDeducer` which allow the most recent emissions
+of a list of deducers to be combined:
+
+```java
+Flux.from(chessGamePublisher)
+    .transform(
+        new ParallelDeducer<>(
+            new ChessStrategyDeducer(),
+            new OtherChessStrategyDeducer(),
+            (chessStrategy, otherChessStrategy) -> {
+              // combine, pick, or create a new strategy and return it
+            }))
+```
+
+As can be seen in the example above, `ParallelDeducer` takes deducers as its first *n* arguments (n <= 10)
+and a combining function as its last argument. The combining function takes *n* arguments, where each
+argument corresponds to the latest emission by the *n*th provided deducer.
+
+### Putting it together
+
+> graphic of how games are modelled with zosma goes here
+
+`zosma` provides a helper class named `Zosma` for modeling game IO in five parts which represents the
+run configuration of the system:
+
+  1. Publisher of external input
+      * The external input to the system, which is represented as a `Publisher` of the raw input to the system.
+  1. Internal input adapter (optional)
+      * The internal input adapter to the system, which transforms data emitted by the external input into a format which can be processed by the game agent.
+  1. Internal output adapter
+      * The internal output adapter to the system, which transforms data emitted by the internal input adapter into a format which can be processed by the internal output adapter. This is where the game logic resides.
+  1. External output adapter (optional)
+      * The external output adapter to the system, which transforms data emitted by the internal output adapter into a format which can be processed by the external output adapter.
+  1. Subscriber
+      * The subscription mechanism of the system, which consumes data emitted by the external output adapter.
+
+The internal input adapter and external output adapter are optional adapters which allow data types
+from external libraries to be converted to an intermediary type which is interpreted by the game agent.
+
+Following the example of a chess game, `Zosma` might be instantiated in the long form as follows:
+
+```java
+final Zosma chessZosma = new Zosma<>(
+    new ChessGamePublisher(),
+    new ChessGameRepresentationAdapter(),
+    new ChessStrategizer(),
+    new ChessStrategyRepresentationAdapter(),
+    new ChessStrategySubscriber());
+```
+
+Or in the short form as follows:
+
+```java
+final Zosma chessZosma = new Zosma<>(
+    new ChessGamePublisher(),
+    new ChessStrategizer(),
+    new ChessStrategySubscriber());
+```
+
+Once instantiated, the game agent can be run by calling `Zosma#run`:
+
+```java
+chessZosma.run();
```

Comment:
It might be intresting to mention that, since `Zosma` implements `Runnable`, it can be ran in a seperate thread, or scheduled in an executorservice :)